CNI Project Security Self-Assessment - Security Pals

jasonliny opened the pull request:
Created and added first draft for CNI Project Security Self-Assessment.
Please feel free to share your feedback on the security self-assessment!

eddie-knight commented on assessments/projects/cni/docs/CNI_SBOM_cyclonedx.bom.json:
I noticed that you included an SBOM along with the self assessment. There are two reasons that jump to the front of my mind for why this isn't needed...

1. SBOMs should be associated with releases, as the bill of materials is only accurate and useful if it is created at build time and associated to a particular point in the code history.
1. If you need to link to an SBOM for some reason in the self-assessment, you can just provide a link out to the latest build artifacts that contain an SBOM.

We still have plenty more to review, but as a starter— could you please remove the SBOM from this PR?

eddie-knight commented on assessments/projects/cni/self-assessment.md:
```diff
@@ -0,0 +1,308 @@
+# Self-assessment
```
```suggestion
# CNI Self-assessment
```

eddie-knight commented on assessments/projects/cni/self-assessment.md:
```diff
@@ -0,0 +1,308 @@
+# Self-assessment
+The Self-assessment is the initial document for projects to begin thinking about the
+security of the project, determining gaps in their security, and preparing any security
+documentation for their users. This document is ideal for projects currently in the
+CNCF **sandbox** as well as projects that are looking to receive a joint assessment and
+currently in CNCF **incubation**.
+
+# Self-assessment outline
```
```suggestion
```

Please go back to ensure that no template text like this is included here or in the rest of the document

jasonliny commented:
Thank you for the initial feedback. 
I have removed the SBOM for the time being since CNI doesn't generate one at the moment and I believe I have removed any preexisting template text.

eddie-knight commented on assessments/projects/cni/self-assessment.md:
Following discussions from the review team, we are asking that Security Pals please remove any recommendations from the scope of this document. You may optionally communicate recommendations for projects in a more accessible location, such as in a project community channel or GitHub Issue.

eddie-knight commented on assessments/projects/cni/self-assessment.md:
```diff
@@ -0,0 +1,277 @@
+
+## Self-assessment use
+
+This self-assessment is created by the CNI team to perform an internal analysis of the
```
Was this created with active participation and review by the CNI team? If not, please remove that reference or replace it with a more accurate statement.

Consider listing the assessment contributors in the topmost paragraph, below the title and before the Table of Contents.

jasonliny commented:
I removed the recommendations, added the authors near the top, and clarified that the self-assessment was made for the CNI project as opposed to being created by the CNI team.

eddie-knight commented on assessments/projects/cni/self-assessment.md:
```diff
@@ -0,0 +1,274 @@
+
+## Overview
+
+**Container Network Interface** <br>
+CNI, a project under the Cloud Native Computing Foundation, includes a set of guidelines and software libraries
```
```suggestion
Container Network Interface (CNI), a project under the Cloud Native Computing Foundation, includes a set of guidelines and software libraries
```

eddie-knight commented on assessments/projects/cni/self-assessment.md:
```diff
@@ -0,0 +1,274 @@
+
+With the rapid development of application containers on Linux, networking in this area is still not well addressed
+as it is highly environment-specific. To deal with this problem, developers seek to create a network layer that is
+pluggable for container runtimes. Ultimately, CNI, along with libraries for Go and a set of plugins, was introduced
```
```suggestion
pluggable for container runtimes. Ultimately, CNI was introduced
```
Consider moving `along with libraries for Go and a set of plugins` to a sentence that can elaborate more specifically on what libraries and plugins you're referencing here.

jasonliny commented:
I have made the corrections to the first sentence in the Overview and in the Background section, added a sentence expanding on how a library is provided for integrating CNI into products and that the team maintains some reference plugins.

eddie-knight commented on assessments/projects/cni/self-assessment.md:
Following discussions among the review team, we are asking that all STRIDE content be moved to a secondary document within this directory. Please extract this content and add a link to the new file in the appendix.

jasonliny replied:
Got it. It has been moved to a separate document and a link to it is provided.

torinvdb commented on assessments/projects/cni/docs/threat-modeling.md:
```diff
@@ -0,0 +1,27 @@
+# CNI Threat Modeling
```
Minor suggestion: consider renaming this to something like threat analysis, threat enumeration, lightweight threat assessment, etc. Presenting it as a threat model might be a bit misleading as that entails a comprehensive process and would be out of scope for the self-assessment (see: [OWASP Threat Modeling Process](https://owasp.org/www-community/Threat_Modeling_Process)).

jasonliny replied:
Thanks for the suggestion. I wasn't aware of the differences in terminology so I really appreciate the feedback!
I've renamed the references of threat modeling to lightweight threat assessment so it's clearer in describing the work we've done.

torinvdb replied:
No worries, looks good overall - thank you!

JustinCappos commented on assessments/projects/cni/self-assessment.md:
```diff
@@ -0,0 +1,249 @@
+Plugins can return one of three result types:
+* Success
+* Error
+* \_Version
```
I wish we had more context here about why these are listed and what the meaning of these is in context...

JustinCappos commented on assessments/projects/cni/docs/lightweight-threat-assessment.md:
```diff
@@ -0,0 +1,27 @@
+# CNI Lightweight Threat Assessment
```
Why is this file included?  (I'm not saying to remove it, but I want to understand why we're adding it as well.)

ragashreeshekar commented on assessments/projects/cni/self-assessment.md:
```diff
@@ -0,0 +1,249 @@
+
+## Self-assessment use
+
+This self-assessment is created for the CNI project to perform an internal analysis of the
+project's security.  It is not intended to provide a security audit of CNI, or
```
```suggestion
This self-assessment is not intended to provide a security audit of CNI, or
```
